internal/ringbuffer.go

```diff
@@ -139,3 +135,35 @@ func (rp *RingbufferProxy) checkCounts(minCount int32, maxCount int32) (err erro
 	}
 	return
 }
+
+type LazyReadResultSet struct {
+	readCount            int32
+	dataItems            []*Data
+	itemSequences        []int64
+	serializationService *SerializationService
+}
+
+const SEQUENCE_UNAVAILABLE int64 = -1
+
+func NewLazyReadResultSet(readCount int32, items []*Data, itemSeqs []int64, ss *SerializationService) *LazyReadResultSet {
+	return &LazyReadResultSet{readCount: readCount, dataItems: items, itemSequences: itemSeqs, serializationService: ss}
+}
+
+func (rs *LazyReadResultSet) ReadCount() int32 {
+	return rs.readCount
+}
+
+func (rs *LazyReadResultSet) Get(index int32) (result interface{}, err error) {
+	return rs.serializationService.ToObject(rs.dataItems[index])
```

Comment:
What if the index is out of boundaries ?

Reply:
we need to add index validation

Reply:
I think we need index validation in the proxies as well but server will return an error in that case but here it will panic so this is more vital I guess.